pkg/apis/imageregistry/v1/types.go

```diff
@@ -108,23 +108,23 @@ type ImageRegistrySpec struct {
 
 	// ReadOnly indicates whether the registry instance should reject attempts
 	// to push new images or delete existing ones.
-	ReadOnly bool `json:"readOnly,omitempty"`
+	ReadOnly bool `json:"readOnly"`
 
 	// Requests controls how many parallel requests a given registry instance will handle before queuing additional requests
 	Requests ImageRegistryConfigRequests `json:"requests,omitempty"`
 
 	// DefaultRoute indicates whether an external facing route for the registry
 	// should be created using the default generated hostname
-	DefaultRoute bool `json:"defaultRoute,omitempty"`
+	DefaultRoute bool `json:"defaultRoute"`
 
 	// Routes defines additional external facing routes which should be created for the registry
 	Routes []ImageRegistryConfigRoute `json:"routes,omitempty"`
 
 	// Replicas determines the number of registry instances to run
-	Replicas int32 `json:"replicas,omitempty"`
+	Replicas int32 `json:"replicas"`
```

Comment:
I'm concerned about this defaulting to 0 - does the operator force this to be at least 1?

Reply:
@adambkaplan No.

Reply:
It doesn't change the default value. It will show the field in `kubectl edit` and `kubectl get -o yaml` even if its value is zero.